src/main/java/com/cburch/logisim/std/ttl/Ttl74192.java

```diff
@@ -0,0 +1,244 @@
+      isPressed = isInside(state, e);
+    }
+
+    @Override
+    public void mouseReleased(InstanceState state, MouseEvent e) {
+      if (!state.getAttributeValue(TtlLibrary.DRAW_INTERNAL_STRUCTURE).booleanValue()) return;
+      if (isPressed && isInside(state, e)) {
+        var index = getIndex(state, e);
+        final var data = (TtlRegisterData) state.getData();
+        if (data == null) return;
+        var current = data.getValue().toLongValue();
+        final long bitValue = 1 << index;
+        current ^= bitValue;
+        updateState(state, current, Value.FALSE, Value.FALSE);
+      }
+      isPressed = false;
+    }
+  }
+
+  @Override
+  public void paintInternal(InstancePainter painter, int x, int y, int height, boolean up) {
+    final var gfx = (Graphics2D) painter.getGraphics();
+    super.paintBase(painter, false, false);
+    Drawgates.paintPortNames(
+        painter,
+        x,
+        y,
+        height,
+        new String[] {
+          "B", "QB", "QA", "CntD", "CntU", "QC", "QD", "D", "C", "LOAD", "CAR", "BOR", "CLR", "A"
+        });
+    TtlRegisterData data = (TtlRegisterData) painter.getData();
+    drawState(gfx, x, y, height, data);
+  }
+
+  private void drawState(Graphics2D gfx, int x, int y, int height, TtlRegisterData state) {
+    if (state != null) {
+      long value = state.getValue().toLongValue();
+      for (var i = 0; i < 4; i++) {
+        final var isSetBitValue = (value & (1 << (3 - i))) != 0;
+        gfx.setColor(isSetBitValue ? trueColor : falseColor);
+        gfx.fillOval(x + 52 + i * 10, y + height / 2 - 4, 8, 8);
+        gfx.setColor(Color.WHITE);
+        GraphicsUtil.drawCenteredText(gfx, isSetBitValue ? "1" : "0", x + 56 + i * 10, y + height / 2);
+      }
+      gfx.setColor(Color.BLACK);
+    }
+  }
+
+  public static void updateState(InstanceState state, Long value, Value carry, Value borrow) {
+    var data = getStateData(state);
+
+    data.setValue(Value.createKnown(BitWidth.create(6), value + (carry.toLongValue() << 4) + (borrow.toLongValue() << 5)));
+    final var vA = data.getValue().get(0);
+    final var vB = data.getValue().get(1);
+    final var vC = data.getValue().get(2);
+    final var vD = data.getValue().get(3);
+    final var vCar = data.getValue().get(4);
+    final var vBor = data.getValue().get(5);
+
+    state.setPort(PORT_INDEX_QA, vA, 4);
+    state.setPort(PORT_INDEX_QB, vB, 4);
+    state.setPort(PORT_INDEX_QC, vC, 4);
+    state.setPort(PORT_INDEX_QD, vD, 4);
+    state.setPort(PORT_INDEX_CARRY, vCar, 4);
+    state.setPort(PORT_INDEX_BORROW, vBor, 4);
+  }
+
+  public static TtlRegisterData getStateData(InstanceState state) {
+    var data = (TtlRegisterData) state.getData();
+    if (data == null) {
+      data = new TtlRegisterData(BitWidth.create(6)); // 4 Data + carry + borrow
+      state.setData(data);
+    }
+    return data;
+  }
+
+  @Override
+  public void propagateTtl(InstanceState state) {
+    var data = getStateData(state);
```

Comment:
Looks like there is few variables in this method can potentially be made `final`. Please check and turn into `final var` whenever possible.